File: day9/day9.go
package day9

import (
	"fmt"
	"os"

	"github.com/tildedave/advent-of-code-2019/intcode"
	"github.com/tildedave/advent-of-code-2019/utils"
)

func Run(f *os.File, partTwo bool) {
	program := utils.ParseProgram(f)
	input := make(chan int)
	output := make(chan int)
	if !partTwo {
		go intcode.ExecFull(program, input, output)
		// test mode
		input <- 1
		for o := range output {
			fmt.Println(o)
		}
	} else {
		go intcode.ExecFull(program, input, output)
		input <- 2
		for o := range output {
			fmt.Println(o)
		}

	}
}

File: src/advent2019/day9.clj
(ns advent2019.day9
  (:require [advent2019.intcode :as intcode]
            [clojure.core.async :as a :refer [>!! <!! <!]]))

;; part 1
(let [input (a/chan)
      output (a/chan)]
 (a/go
   (intcode/run-program (intcode/parse-file "2019/day9.txt")
                      input output))
  (>!! input 1)
  (println (last (<!! (a/into [] output)))))

;; part 2
(let [input (a/chan)
      output (a/chan)]
 (a/go
   (intcode/run-program (intcode/parse-file "2019/day9.txt")
                      input output))
  (>!! input 2)
  (println (last (<!! (a/into [] output)))))
